File: leetcode/tree/binary_tree_inorder_traversal.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

var result []int

func inorderTraversal(root *TreeNode) {
	if root == nil {
		return
	}
	inorderTraversal(root.Left)
	result = append(result, root.Val)
	inorderTraversal(root.Right)
}

func main() {
	root := &TreeNode{Val: 1, Left: nil, Right: nil}
	root.Left = &TreeNode{Val: 2, Left: nil, Right: nil}
	root.Right = &TreeNode{Val: 3, Left: nil, Right: nil}
	root.Right.Left = &TreeNode{Val: 4, Left: nil, Right: nil}
	inorderTraversal(root)
	for _, val := range result {
		fmt.Printf("%d ", val)
	}
	fmt.Println()
}

File: leetcode/tree/binary_tree_inorder_traversal.py
from common import TreeNode

def inorderTraversal(root: TreeNode) -> list:
    result = []
    def solve(root: TreeNode):
        if not root:
            return
        solve(root.left)
        result.append(root.val)
        solve(root.right)
        return result
    return solve(root)

'''
    1
   / \
  2   3
     /
    4 
=> inorder is: 2 - 1 - 4 - 3 
'''
if __name__ == "__main__":
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.right.left = TreeNode(4)
    print("Inorder is: {}".format(inorderTraversal(root)))